components/merkledb/src/macros.rs

```diff
@@ -152,3 +152,47 @@ macro_rules! impl_serde_hex_for_binary_value {
         }
     };
 }
+
+/// Implements `BinaryValue` traint for tuples cross single member generic tuple struct.
+/// If expression begins with for keyword it creates implementation for existend structure
+/// supposing it meets requirements, mentioned above. Otherwise it creates new structure
+/// according to supplied meta attributes, visibility, name and type constraint for generic type.
+/// Target tuple types must be listed in curly braces.
+#[macro_export]
+macro_rules! binary_value_tuple_impls {
```

Comment:
I don't understand why this macro needs to be public, or why it needs to be defined in the `merkledb` crate. See the overall comment for more details.